solidity/random-beacon/test/RandomBeacon.Relay.test.ts

```diff
@@ -850,10 +766,569 @@ describe("RandomBeacon - Relay", () => {
+                // is expected to be provided by signer 33.
+                const { signatures, signingMembersIndices } =
+                  await signHeartbeatFailureClaim(
+                    members,
+                    0,
+                    group.groupPubKey,
+                    validFailedMembersIndices,
+                    groupThreshold - 1
+                  )
+
+                await expect(
+                  randomBeacon.notifyFailedHeartbeat({
+                    groupId,
+                    failedMembersIndices: validFailedMembersIndices,
+                    // Slice removes `0x` prefix from wrong signature.
+                    signatures: signatures + invalidSignature.slice(2),
+                    signingMembersIndices: [...signingMembersIndices, 33],
+                  })
+                ).to.be.revertedWith("Invalid signature")
+              }
+
+              context(
+                "when one of the signatures signed the wrong nonce",
+                () => {
+                  it("should revert", async () => {
+                    // Signer 33 signs wrong nonce.
+                    const invalidSignature = (
+                      await signHeartbeatFailureClaim(
+                        [members[32]],
+                        1,
+                        group.groupPubKey,
+                        validFailedMembersIndices,
+                        1
+                      )
+                    ).signatures
+
+                    await assertInvalidSignature(invalidSignature)
+                  })
+                }
+              )
+
+              context(
+                "when one of the signatures signed the wrong group public key",
+                () => {
+                  it("should revert", async () => {
+                    // Signer 33 signs wrong group public key.
+                    const invalidSignature = (
+                      await signHeartbeatFailureClaim(
+                        [members[32]],
+                        0,
+                        "0x010203",
+                        validFailedMembersIndices,
+                        1
+                      )
+                    ).signatures
+
+                    await assertInvalidSignature(invalidSignature)
+                  })
+                }
+              )
+
+              context(
+                "when one of the signatures signed the wrong failed group members indices",
+                () => {
+                  it("should revert", async () => {
+                    // Signer 33 signs wrong failed group members indices.
+                    const invalidSignature = (
+                      await signHeartbeatFailureClaim(
+                        [members[32]],
+                        0,
+                        group.groupPubKey,
+                        [1, 2, 3, 4, 5, 6, 7, 8],
+                        1
+                      )
+                    ).signatures
```

Comment:
Can we define `constants.sortitionPoolRewardsBanDuration` in `fixtures/index.ts` and use it here?

```ts
const expectedUntil = now + constants.sortitionPoolRewardsBanDuration
```

Reply:
See https://github.com/keep-network/keep-core/pull/2736/commits/2862df871e16cbcc26fb40fbb6f1d7b408746834. Extracted to `params` though since it's a governable thing.